curve_apps/utils.py

```diff
@@ -0,0 +1,172 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of edge-detection package.
+#
+#  All rights reserved.
+#
+
+from __future__ import annotations
+
+import numpy as np
+from scipy.spatial import Delaunay
+
+from curve_apps.trend_lines.params import DetectionParameters
+
+
+def find_curves(
```

Comment:
Thoses functions are quite complexe. 
Cutting them in sub function could help clarity

Reply:
let's do it here. Need those changes to go forward for other issues
https://mirageoscience.atlassian.net/browse/GEOPY-1537